packages/story-editor/src/components/library/libraryProvider.js

```diff
@@ -54,11 +54,22 @@ const LIBRARY_TAB_IDS = new Set(
 );
 
 function LibraryProvider({ children }) {
-  const { showMedia3p, canViewDefaultTemplates } = useConfig();
+  const { showMedia3p, shoppingProvider, canViewDefaultTemplates } =
+    useConfig();
   const {
     actions: { getMedia, getCustomPageTemplates },
   } = useAPI();
-  const isShoppingEnabled = useFeature('shoppingIntegration');
+
+  const { currentPageProducts } = useStory(({ state: { currentPage } }) => ({
+    currentPageProducts: currentPage?.elements?.filter(
+      ({ type }) => type === 'product'
+    ),
+  }));
```

Comment:
Since we're only interested in the number of products, better to return a bool here instead of the whole list of products. Helps reduce unnecessary renders.

Something like this:

```suggestion
  const { hasProducts } = useStory(({ state: { currentPage } }) => ({
    hasProducts: currentPage?.elements?.some(
      ({ type }) => type === 'product'
    ),
  }));
```

Reply:
Fixed.